src/main/scala/mesosphere/marathon/api/akkahttp/EntityMarshallers.scala

```diff
@@ -95,10 +96,49 @@ object EntityMarshallers {
     jsonStringMarshaller.compose(printer).compose(internal => writes.writes(ramlWrites.write(internal)))
 
   import mesosphere.marathon.raml.AppConversion.appRamlReader
-  implicit def appUnmarshaller(
+  implicit def appDefinitionUnmarshaller(
     implicit
-    normalization: Normalization[raml.App], validator: Validator[AppDefinition]) = {
-    validEntityRaml(playJsonUnmarshaller[raml.App])
+    normalization: Normalization[raml.App],
+    validator: Validator[AppDefinition]): FromEntityUnmarshaller[AppDefinition] = {
+    validEntityRaml(playJsonUnmarshaller[raml.App]).handleValidationErrors
+  }
+
+  implicit val appDefinitionMarshaller: ToEntityMarshaller[AppDefinition] =
+    playJsonMarshaller[raml.App].compose { appDefinition: AppDefinition =>
+      raml.Raml.toRaml(appDefinition)
+    }
+
+  def appUpdateUnmarshaller(
+    appId: PathId, partialUpdate: Boolean)(
+    implicit
+    appUpdateNormalization: Normalization[raml.AppUpdate],
+    appNormalization: Normalization[raml.App]): FromEntityUnmarshaller[raml.AppUpdate] = {
+    if (partialUpdate) {
+      playJsonUnmarshaller[raml.AppUpdate].map { appUpdate =>
+        appUpdateNormalization.normalized(appUpdate.copy(id = Some(appId.toString)))
```

Comment:
Hm, I wonder if the serialization/deserialization if the right spot for a normalization. To me it feels this should be done after serialization. What do you think?